src/cljs/athens/events.cljs

```diff
@@ -1698,28 +1699,34 @@
                                         embed-id (str "-embed-" embed-id)) n]))]})))
 
 
+(defn- waiting-for-ack [db event-id]
+  (update db :ui/waiting-for-ack #(conj (or % #{}) event-id)))
+
+
+(reg-event-fx
+ :remote/paste-verbatim
+ (fn [{db :db} [_ uid text start value]]
+   (let [last-seen-tx         1 ;; TODO last-seen-tx discovery
+         event-id             (gensym)
+         paste-verbatim-event {:event/id      event-id     ;; use `:event/id` to track `:ack` events
+                               :event/last-tx last-seen-tx ;; in case if event could conflict and was issued from not up to date db
+                               :event/type    :apaste-verbatim
+                               :event/args    {:uid   uid
+                                               :text  text
+                                               :start start
+                                               :value value}}]
+     {:db                 (waiting-for-ack db event-id)
+      :remote/send-event! paste-verbatim-event})))
+
 (reg-event-fx
  :paste-verbatim
- (fn [_ [_ uid text]]
+ (fn [{db :db} [_ uid text]]
    (let [{:keys [start value]} (textarea-keydown/destruct-target js/document.activeElement)
-         block-empty?          (string/blank? value)
-         block-start?          (zero? start)
-         new-string            (cond
-
-                                 block-empty?
-                                 text
-
-                                 (and (not block-empty?)
-                                      block-start?)
-                                 (str text value)
-
-                                 :else
-                                 (str (subs value 0 start)
-                                      text
-                                      (subs value start)))
-         tx-data [{:db/id        [:block/uid uid]
-                   :block/string new-string}]]
-     {:dispatch [:transact tx-data]})))
+         local?                (not (:db/remote-graph-conf db))]
+     (if local?
+       {:dispatch [:transact (common-events/paste-verbatim->tx uid text start value)]}
+       ;; TODO event handler for `:remote/paste-verbatim`
+       {:dispatch [:remote/paste-verbatim uid text start value]}))))
```

Comment:
So every datascript+re-frame transaction needs to check if `local?`?

Is this a good use case for [interceptors](https://day8.github.io/re-frame/Interceptors/)? Every `ds+rf` event can have a `local` interceptor that decides to write transaction on client datascript or through backend datahike.

Reply:
I need to check, haven't had the need for using interceptors in past.

Reply:
Ok, checked.
Using interceptors to "route" events would be possible (through self modifying property of interceptors)
but it would be confusing, because we'd register event-handler `:paster-verbatim` with this magic interceptor,
and magic interceptor would re-route event to other handler, this would result in very unintuitive code.

We could inject cofx to test if we're in local vs self-hosted mode.

Reply:
Cofx is probably the right move then! Both Cofx and Interceptors are good at reusing some code for events, but Interceptors are a little too magical.